python/triton/compiler/compiler.py

```diff
@@ -45,7 +45,12 @@ def ttir_compute_capability_rewrite(mod, arch):
     pm = ir.pass_manager(mod.context)
     pm.enable_debug()
     if _is_cuda(arch):
-        pm.add_rewrite_tensor_pointer_pass(arch)
+        pm.add_rewrite_tensor_pointer_pass(arch, False)
+    else:
+        device = get_current_device()
+        capabilityTuple = get_device_capability(device)
+        capability = capabilityTuple[0] * 10 + capabilityTuple[1]
+        pm.add_rewrite_tensor_pointer_pass(capability, True)
```

Comment:
Not sure compute capability make sense for non-nvidia architecrtures, maybe do something like this?

```suggestion
    elif is_hip():
        capability = 90
        pm.add_rewrite_tensor_pointer_pass(capability, True)
    else:
        assert(False, "unsupported target")
```